### PR TITLE
fix(SF-997): shouldUpdate function needs to support forced updates

### PR DESCRIPTION
**Pull request by Vic-Dev**

### CODE REVIEW

File: src/core/tag/mixins/pure.ts [hunk 1]
```diff
@@ -1,161 +1,161 @@
...
+      prevProps = tag.props;
+      prevAliases = Pure.extractLocalAliases(tag);
+    };
+
+    tag.one(Phase.MOUNT, updatePrev);
+    tag.on(Phase.UPDATED, updatePrev);
+
+    return (stateChange: object | true, nextOpts: object) => {
+      const nextProps = Props.buildProps(tag, nextOpts);
+      const forceUpdate = stateChange === true;
```
Comment (jmbtutor) [inline src/core/tag/mixins/pure.ts]: `stateChange === true || !('state' in stateChange)`

File: src/core/tag/mixins/pure.ts [hunk 2]
```diff
@@ -1,161 +1,161 @@
...
+      const nextState = stateChange != null ? { ...tag.state, ...stateChange['state'] } : tag.state;
```
Comment (jmbtutor) [inline src/core/tag/mixins/pure.ts]: Needs to handle `stateChange === true`.

File: test/unit/core/tag/mixins/pure.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    it('should consider objects with extra properties as not equal', () => {
+      const lhs = { a: 'b', c: 'd' };
+      const rhs = { a: 'b' };
+
+      expect(Pure.shallowEquals(lhs, rhs)).to.be.false;
+    });
+  });
+
+  describe('composeObject()', () => {
+    it('should extract the values for a given key and make a new object', () => {
```
Comment (jmbtutor) [inline test/unit/core/tag/mixins/pure.ts]: `for the given keys`

File: src/core/tag/mixins/pure.ts (hunk 1 above)
Comment (Vic-Dev) [reply src/core/tag/mixins/pure.ts]: no

**Pull request merged**